### PR TITLE
[ISSUE #8694] Support prometheus http service discovery

### PR DESCRIPTION
**Pull request by karsonto**
Fix Issue #8694
add enable to support prometheus service discovery config in application.properties
add prometheus service discovery uri config in application.properties

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: console/src/main/resources/application.properties [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #*************** Spring Boot Related Configurations ***************#
 ### Default web context path:
 server.servlet.contextPath=/nacos
+### Setting Content-Type response header to include ;charset=UTF-8
+server.servlet.encoding.force-response=true
```
Comment (KomachiSion) [inline console/src/main/resources/application.properties:21]: Is this config will affect other api response?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: console/src/main/resources/application.properties (hunk 1 above)
Comment (karsonto) [reply console/src/main/resources/application.properties:21]: 根据官网的解析是强制在response content-type上加上字符集，不会对现有api造成影响
image

**Comment by KomachiSion** [comment]
强制指定字符集可能会对配置中心有影响吧 @realJackSun @shiyiyue1102 看下

**Comment by karsonto** [comment]
> 强制指定字符集可能会对配置中心有影响吧 @realJackSun @shiyiyue1102 看下

如果不加这个更加的奇怪，字符集是iso8859-1 ，加上这个才受控制

**Comment by KomachiSion** [comment]
> > 强制指定字符集可能会对配置中心有影响吧 @realJackSun @shiyiyue1102 看下
> 
> 如果不加这个更加的奇怪，字符集是iso8859-1 ，加上这个才受控制

现在好像配置中心默认的返回编码都是utf-8，但是某些环境下会使用gbk， 如果这里强制使用utf-8， 返回的配置中文可能乱码，导致程序会有错误。

**Comment by karsonto** [comment]
er....那我先用其他方法代替，还是等待 @realJackSun @shiyiyue1102 评估一下？

**Review by realJackSun — APPROVED** [review]

**Comment by KomachiSion** [comment]
> er....那我先用其他方法代替，还是等待 @realJackSun @shiyiyue1102 评估一下？

先尝试找下其他办法吧。

**Comment by KomachiSion** [comment]
Closed due to #9518

**Pull request closed without merging**